core/src/main/java/org/apache/iceberg/SchemaParser.java

```diff
@@ -145,6 +145,8 @@ static void toJson(Type.PrimitiveType primitive, JsonGenerator generator) throws
   static void toJson(Type type, JsonGenerator generator) throws IOException {
     if (type.isPrimitiveType()) {
       toJson(type.asPrimitiveType(), generator);
+    } else if (type.isVariantType()) {
+      generator.writeString(type.toString());
```

Comment:
I'd probably just inline the `toJson` call from the previous case along with this:

```java
    if (type.isPrimitiveType() || type.isVariantType()) {
      generator.writeString(type.toString());
    } else {
```